airflow/triggers/temporal.py

```diff
@@ -34,18 +35,29 @@ class DateTimeTrigger(BaseTrigger):
     The provided datetime MUST be in UTC.
     """
 
-    def __init__(self, moment: datetime.datetime):
+    def __init__(self, moment: datetime.datetime, soft_fail: bool = False):
         super().__init__()
         if not isinstance(moment, datetime.datetime):
-            raise TypeError(f"Expected datetime.datetime type for moment. Got {type(moment)}")
+            exc = TypeError(f"Expected datetime.datetime type for moment. Got {type(moment)}")
+            if soft_fail:
+                raise AirflowSkipException("Skipping due to soft_fail is set to True.") from exc
+            raise exc
```

Comment:
Do we need to use `from` here? I feel it’d be more concise to combine the two error messages into `AirflowSkipException` instead.

Reply:
`from` is not necessary here. Either way works fine for me. Let me change to what you suggested. 🙂

Reply:
Let’s raise the skip exception directly (and include the type error message in it)

Reply:
Sure. just updated it!